terraform-import-github-org.sh

```diff
@@ -78,99 +138,32 @@ import_private_repos () {
 # Users
 import_users () {
   for i in $(curl -s "${API_URL_PREFIX}/orgs/$ORG/members?access_token=$GITHUB_TOKEN&per_page=100" | jq -r 'sort_by(.login) | .[] | .login'); do
- 
-  echo "${API_URL_PREFIX}/orgs/$ORG/members?access_token=$GITHUB_TOKEN&per_page=100"
-
-  TEAM_ROLE=$(curl -s "${API_URL_PREFIX}/orgs/$ORG/memberships/$i?access_token=$GITHUB_TOKEN&per_page=100" -H "Accept: application/vnd.github.hellcat-preview+json" | jq -r .role)
-
-  cat >> github-users.tf << EOF
-resource "github_membership" "$i" {
-  username        = "$i"
-  role            = "$TEAM_ROLE"
-}
-EOF
+    declare_github_membership "$i" >> github-users.tf
     terraform import github_membership.$i $ORG:$i
   done
 }
 
 # Teams
 import_teams () {
+  local results
+  local tempfile=github-teams.recent-slug
   for i in $(curl -s "${API_URL_PREFIX}/orgs/$ORG/teams?access_token=$GITHUB_TOKEN&per_page=100" -H "Accept: application/vnd.github.hellcat-preview+json" | jq -r 'sort_by(.name) | .[] | .id'); do
-  
-    TEAM_NAME=$(curl -s "${API_URL_PREFIX}/teams/$i?access_token=$GITHUB_TOKEN&per_page=100" -H "Accept: application/vnd.github.hellcat-preview+json" | jq -r .name)
-
-    TEAM_SLUG=$(curl -s "${API_URL_PREFIX}/teams/$i?access_token=$GITHUB_TOKEN&per_page=100" -H "Accept: application/vnd.github.hellcat-preview+json" | jq -r .slug)
-
-    TEAM_PRIVACY=$(curl -s "${API_URL_PREFIX}/teams/$i?access_token=$GITHUB_TOKEN&per_page=100" -H "Accept: application/vnd.github.hellcat-preview+json" | jq -r .privacy)
-  
-    TEAM_DESCRIPTION=$(curl -s "${API_URL_PREFIX}/teams/$i?access_token=$GITHUB_TOKEN&per_page=100" -H "Accept: application/vnd.github.hellcat-preview+json" | jq -r .description)
-
-    TEAM_PARENT=$(curl -s "${API_URL_PREFIX}/teams/$i?access_token=$GITHUB_TOKEN&per_page=100" -H "Accept: application/vnd.github.hellcat-preview+json" | jq -r .parent.id)
-    if [[ "$TEAM_PARENT" == "null" ]]; then
-      TEAM_PARENT=""
-    fi
-
-    if [[ "$TEAM_PRIVACY" == "closed" ]]; then
-      cat >> github-teams-$TEAM_SLUG.tf << EOF
-resource "github_team" "$TEAM_SLUG" {
-  name        = "$TEAM_NAME"
-  description = "$TEAM_DESCRIPTION"
-  privacy     = "closed"
-  parent_team_id = "$TEAM_PARENT"
-}
-EOF
-    elif [[ "$TEAM_PRIVACY" == "secret" ]]; then
-      cat >> github-teams-$TEAM_SLUG.tf << EOF
-resource "github_team" "$TEAM_SLUG" {
-  name        = "$TEAM_NAME"
-  description = "$TEAM_DESCRIPTION"
-  privacy     = "secret"
-  parent_team_id = "$TEAM_PARENT"
-}
-EOF
+    if ! declare_github_team "${i}" > "${tempfile}" ; then
```

Comment:
We should be okay with the change from `>>` to `>` here given that each team has one entry in one file, no?